ipfailover_keepalived.yaml

```diff
@@ -0,0 +1,58 @@
+heat_template_version: 2014-10-16
+
+description: >
+  A template which deploys a IP failover service for the Openshift router
+
+parameters:
+  fixed_network:
+    description: >
+      The name or ID of the admin and public network
+    type: string
+    constraints:
+    - custom_constraint: neutron.network
+
+  fixed_subnet:
+    description: >
+      The name or ID of the admin and public IPv4 space
+    type: string
+    constraints:
+    - custom_constraint: neutron.subnet
+
+  external_network:
+    description: >
+      The name of the inbound access network
+    type: string
+    constraints:
+    - custom_constraint: neutron.network
+
+resources:
+  router_vip_port:
+    type: OS::Neutron::Port
+    properties:
+      network_id: {get_param: fixed_network}
+      fixed_ips:
+       - subnet: {get_param: fixed_subnet}
+
+  router_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: {get_param: external_network}
+      port_id: {get_resource: router_vip_port}
+
+outputs:
+  router_vip:
+    description: >
+      The Openshift Router Virtual IP
+    value: {get_attr: [router_vip_port, fixed_ips, 0, ip_address]}
```

Comment:
an empty string is returned, this change worked for me:
-    value: {get_attr: [router_vip_port, fixed_ips, 0, ip_address]}
+    value: {get_attr: [router_floating_ip, floating_ip_address]}